File: crates/test-support/src/integration_tests/misc/disabled_keybindings.rs
use crate::integration_test::{IntegrationTest, NewIntegrationTestArgs};

pub struct DisabledKeybindingsTest;

impl DisabledKeybindingsTest {
    pub fn new() -> IntegrationTest {
        IntegrationTest::new(NewIntegrationTestArgs {
            description: "Confirms you can disable keybindings by setting them to <disabled>"
                .to_string(),
            extra_cmd_args: vec![],
            skip: false,
            setup_repo: Some(Box::new(|_shell| {})),
            setup_config: Some(Box::new(|_cfg| {})),
            run: Some(Box::new(|_t, _keys| {})),
            extra_env_vars: std::collections::HashMap::new(),
            git_version: Default::default(),
            width: 0,
            height: 0,
            is_demo: false,
        })
    }
}

pub const DISABLED_KEYBINDINGS: DisabledKeybindingsTest = DisabledKeybindingsTest;

File: pkg/integration/tests/misc/disabled_keybindings.go
package misc

import (
	"github.com/quangdang46/slg/pkg/config"
	. "github.com/quangdang46/slg/pkg/integration/components"
)

var DisabledKeybindings = NewIntegrationTest(NewIntegrationTestArgs{
	Description:  "Confirms you can disable keybindings by setting them to <disabled>",
	ExtraCmdArgs: []string{},
	Skip:         false,
	SetupConfig: func(config *config.AppConfig) {
		config.GetUserConfig().Keybinding.Universal.PrevItem = "<disabled>"
		config.GetUserConfig().Keybinding.Universal.NextItem = "<disabled>"
		config.GetUserConfig().Keybinding.Universal.NextTab = "<up>"
		config.GetUserConfig().Keybinding.Universal.PrevTab = "<down>"
	},
	SetupRepo: func(shell *Shell) {},
	Run: func(t *TestDriver, keys config.KeybindingConfig) {
		t.Views().Files().
			IsFocused().
			Press("<up>")

		t.Views().Worktrees().IsFocused()
	},
})
